core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentUpdateStatusManager.java

```diff
@@ -269,9 +270,9 @@ public ICarbonLock getTableUpdateStatusLock() {
    * @throws Exception
    */
   public String[] getDeleteDeltaFilePath(String blockFilePath) throws Exception {
-    int tableFactPathLength = CarbonStorePath
-        .getCarbonTablePath(absoluteTableIdentifier.getStorePath(),
-            absoluteTableIdentifier.getCarbonTableIdentifier()).getFactDir().length() + 1;
+    String factTableDir =
+        absoluteTableIdentifier.getCarbonTableIdentifier().getTableName() + File.separator + "Fact";
+    int tableFactPathLength = blockFilePath.indexOf(factTableDir) + factTableDir.length() + 1;
```

Comment:
Because when I create table, the location in table properties is 'hdfs:////user/xxxxx/fact', but the fact dir is 'hdfs://cluster/user/xxxxx/fact', so getFactDir().length() + 1 is wrong.